### ISSUE TITLE
[a11y]: Export Modal, Dialog heading is provided in h3 instead of h2

### ISSUE THREAD

**Issue by devadula-nandan** [carbon-design-system/ibm-products#6019]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

v2.49.0-rc.0

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Reported by Raghu during screen reader testing.

Maintain hierarchical order for the Heading levels :
Recommended to present the Modal Main heading in H2.

If page headings are not in the right sequence, screen reader users find it difficult to comprehend the hierarchical structure of the page content. 

### WCAG 2.1 Violation

1.3.1 Info & Relationships

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-export-exportmodal--standard&globals=viewport:basic

### Steps to reproduce

noticeable in storybook

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by anamikaanu96** [comment]
In export modal, we are using composed modal from carbon. This heading h3 is coming from there, I have raised the [issue](https://github.com/carbon-design-system/carbon/issues/17397) in carbon.
